docs/guides/divergence.md

```diff
@@ -0,0 +1,89 @@
+# Handling divergent changes
+
+## What are divergent changes?
+
+A [divergent change][glossary_divergent_change] occurs when multiple visible
+commits have the same change ID.
+
+Normally, when commits are rewritten, the original version (the "predecessor")
+becomes hidden and the new commit (the "successor") is visible. Thus, only one
+commit with a given change ID is visible at a time.
+
+But, a hidden commit can become visible again. This can happen if:
+
+- A visible descendant is added. For example, `jj new REV` will make `REV`
+  visible even if it was hidden before.
+
+- It is made the working copy. `jj edit REV` will make `REV` visible if it
+  wasn't already.
```

Comment:
In 97a57fe7027480f3898fec4e65ebee74bac71bc7:

We should add the case where a new ref is imported from git, e.g. by the user running `git fetch` or otherwise directly manipulating the git repo. For me this is the most common source of divergent commits, and it's especially confusing because "I didn't do anything, I just `git fetch`ed!", which in git is one of the least destructive things you can do.

Reply:
This sounds great to me! I didn't realize that this was the specific mechanism by which my fetches were causing me grief.

I kinda think that `jj` itself should help with this. by noticing when previously-hidden commits become visible again. (And distinguishing between commits that were hidden manually with `abandon` and those that were hidden implicitly by just updating a change). Though this is out of scope for a docs change, of course.